### ISSUE TITLE
Error after installing gem

### ISSUE THREAD

**Issue by Patrax** [deanperry/shoppe-paypal#2]
Much like Error#1, but the solution provided by one of the comments doesn't work and the stack trace is different.

Trying to run the server and getting:

```
Exiting
/usr/local/rvm/gems/ruby-2.1.5@rails4/gems/sqlite3-1.3.10/lib/sqlite3/database.rb:91:in `initialize': SQLite3::SQLException: no such table: shoppe_settings: SELECT "shoppe_settings".* FROM "shoppe_settings" (ActiveRecord::StatementInvalid)
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/sqlite3-1.3.10/lib/sqlite3/database.rb:91:in `new'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/sqlite3-1.3.10/lib/sqlite3/database.rb:91:in `prepare'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/connection_adapters/sqlite3_adapter.rb:301:in `block in exec_query'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/connection_adapters/abstract_adapter.rb:373:in `block in log'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activesupport-4.1.7/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/connection_adapters/abstract_adapter.rb:367:in `log'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/connection_adapters/sqlite3_adapter.rb:298:in `exec_query'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/connection_adapters/sqlite3_adapter.rb:510:in `select'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/connection_adapters/abstract/database_statements.rb:24:in `select_all'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/connection_adapters/abstract/query_cache.rb:70:in `select_all'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/querying.rb:39:in `find_by_sql'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/relation.rb:611:in `exec_queries'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/relation.rb:493:in `load'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/relation.rb:238:in `to_a'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/relation/delegation.rb:132:in `method_missing'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activerecord-4.1.7/lib/active_record/relation/delegation.rb:99:in `method_missing'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/shoppe-1.0.5/app/models/shoppe/setting.rb:44:in `to_hash'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/shoppe-1.0.5/lib/shoppe.rb:29:in `settings'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/shoppe-paypal-1.0.0/lib/shoppe/paypal.rb:36:in `setup'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/shoppe-paypal-1.0.0/lib/shoppe/paypal/engine.rb:6:in `block in <class:Engine>'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/initializable.rb:30:in `instance_exec'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/initializable.rb:30:in `run'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/initializable.rb:55:in `block in run_initializers'
    from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:226:in `block in tsort_each'
    from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
    from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:427:in `each_strongly_connected_component_from'
    from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:347:in `block in each_strongly_connected_component'
    from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:345:in `each'
    from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:345:in `call'
    from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:345:in `each_strongly_connected_component'
    from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:224:in `tsort_each'
    from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:205:in `tsort_each'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/initializable.rb:54:in `run_initializers'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/application.rb:300:in `initialize!'
    from /home/ubuntu/workspace/config/environment.rb:5:in `<top (required)>'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:247:in `require'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:247:in `block in require'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:232:in `load_dependency'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:247:in `require'
    from /home/ubuntu/workspace/config.ru:3:in `block in <main>'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/builder.rb:55:in `instance_eval'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/builder.rb:55:in `initialize'
    from /home/ubuntu/workspace/config.ru:in `new'
    from /home/ubuntu/workspace/config.ru:in `<main>'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/builder.rb:49:in `eval'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/builder.rb:49:in `new_from_string'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/builder.rb:40:in `parse_file'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/server.rb:277:in `build_app_and_options_from_config'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/server.rb:199:in `app'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/server.rb:50:in `app'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/server.rb:314:in `wrapped_app'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/server.rb:130:in `log_to_stdout'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/server.rb:67:in `start'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:81:in `block in server'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:76:in `tap'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:76:in `server'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:40:in `run_command!'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands.rb:17:in `<top (required)>'
    from /home/ubuntu/workspace/bin/rails:8:in `require'
    from /home/ubuntu/workspace/bin/rails:8:in `<top (required)>'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/client/rails.rb:27:in `load'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/client/rails.rb:27:in `call'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/client/command.rb:7:in `call'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/client.rb:26:in `run'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/bin/spring:48:in `<top (required)>'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/binstub.rb:11:in `load'
    from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/binstub.rb:11:in `<top (required)>'
    from /home/ubuntu/workspace/bin/spring:13:in `require'
    from /home/ubuntu/workspace/bin/spring:13:in `<top (required)>'
    from bin/rails:3:in `load'
    from bin/rails:3:in `<main>'   
```

**Comment by deanpcmad** [comment]
What version of the shoppe gem are you using?

On Tuesday, April 28, 2015, Patrax notifications@github.com wrote:

> Much like Error#1, but the solution provided by one of the comments
> doesn't work and the stack trace is different.
> 
> Trying to run the server and getting:
> 
> Exiting
> /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/shoppe-paypal-1.0.0/lib/shoppe/paypal.rb:26:in `setup': undefined method`add_settings_group' for Shoppe:Module (NoMethodError)
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/shoppe-paypal-1.0.0/lib/shoppe/paypal/engine.rb:6:in `block in <class:Engine>'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/initializable.rb:30:in`instance_exec'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/initializable.rb:30:in `run'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/initializable.rb:55:in`block in run_initializers'
>     from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:226:in `block in tsort_each'
>     from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:348:in`block (2 levels) in each_strongly_connected_component'
>     from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:427:in `each_strongly_connected_component_from'
>     from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:347:in`block in each_strongly_connected_component'
>     from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:345:in `each'
>     from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:345:in`call'
>     from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:345:in `each_strongly_connected_component'
>     from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:224:in`tsort_each'
>     from /usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/tsort.rb:205:in `tsort_each'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/initializable.rb:54:in`run_initializers'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/application.rb:300:in `initialize!'
>     from /home/ubuntu/workspace/config/environment.rb:5:in`<top (required)>'
>     from /home/ubuntu/workspace/config.ru:3:in `require'
>     from /home/ubuntu/workspace/config.ru:3:in`block in <main>'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/builder.rb:55:in `instance_eval'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/builder.rb:55:in`initialize'
>     from /home/ubuntu/workspace/config.ru:in `new'
>     from /home/ubuntu/workspace/config.ru:in`<main>'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/builder.rb:49:in `eval'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/builder.rb:49:in`new_from_string'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/builder.rb:40:in `parse_file'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/server.rb:277:in`build_app_and_options_from_config'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/server.rb:199:in `app'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/server.rb:50:in`app'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/rack-1.5.2/lib/rack/server.rb:314:in `wrapped_app'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/server.rb:130:in`log_to_stdout'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/server.rb:67:in `start'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:81:in`block in server'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:76:in `tap'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:76:in`server'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:40:in `run_command!'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/railties-4.1.7/lib/rails/commands.rb:17:in`<top (required)>'
>     from /home/ubuntu/workspace/bin/rails:8:in `require'
>     from /home/ubuntu/workspace/bin/rails:8:in`<top (required)>'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/client/rails.rb:27:in `load'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/client/rails.rb:27:in`call'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/client/command.rb:7:in `call'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/client.rb:26:in`run'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/bin/spring:48:in `<top (required)>'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/binstub.rb:11:in`load'
>     from /usr/local/rvm/gems/ruby-2.1.5@rails4/gems/spring-1.3.3/lib/spring/binstub.rb:11:in `<top (required)>'
>     from /home/ubuntu/workspace/bin/spring:13:in`require'
>     from /home/ubuntu/workspace/bin/spring:13:in `<top (required)>'
>     from bin/rails:3:in`load'
>     from bin/rails:3:in `<main>'
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/deanperry/shoppe-paypal/issues/2.

## 

Sent from Gmail Mobile

**Comment by Patrax** [comment]
Oh, I thought it would import its dependency. Either way, now adding 'shoppe', '~> 1.0.5' I get a different error. I'll update the issue to reflect the new stack trace.

Update: Following shoppe setting up tutorial and will let you know if I run into trouble. Thank you :).

**Comment by Patrax** [comment]
Ok, after following http://tryshoppe.com/docs/tutorials/setting-up I'm getting the same as issue#1. I'll give it a shot again. :)

**Comment by deanpcmad** [comment]
OK. I'm still trying to diagnose that issue but I've just thought of something that may fix it...

**Comment by deanpcmad** [comment]
This is now fixed in v1.1.1

**Comment by deanpcmad** [comment]
I have also updated the documentation for a change in the controller code required - http://tryshoppe.com/docs/payment-gateways/paypal
